ts_src/psbt.ts

```diff
@@ -213,6 +219,82 @@ export class Psbt {
     return this;
   }
 
+  addIssuance(args: AddIssuanceArgs, inputIndex?: number): this {
+    // check the amounts.
+    if (args.assetAmount <= 0)
+      throw new Error('asset amount must be greater than zero.');
+    if (args.tokenAmount < 0) throw new Error('token amount must be positive.');
+
+    if (inputIndex) {
+      // check if the input exists
+      if (!this.data.inputs[inputIndex])
+        throw new Error(`The input ${inputIndex} does not exist.`);
+      // check if the input is available for issuance.
+      if (this.__CACHE.__TX.ins[inputIndex].issuance)
+        throw new Error(`The input ${inputIndex} already has issuance data.`);
+    } else {
+      // verify if there is at least one input available.
+      if (this.__CACHE.__TX.ins.filter(i => !i.issuance).length === 0)
+        throw new Error(
+          'transaction needs at least one input without issuance data.',
+        );
+      // search and extract the input index.
+      inputIndex = this.__CACHE.__TX.ins.findIndex(i => !i.issuance);
+    }
+
+    const { hash, index } = this.__CACHE.__TX.ins[inputIndex];
+
+    // create an issuance object using the vout and the args
+    const issuance: Issuance = newIssuance(
+      args.assetAmount,
+      args.tokenAmount,
+      args.precision,
+      args.contract,
+    );
+
+    // generate the entropy
+    const entropy: Buffer = generateEntropy(
+      { txHash: hash, vout: index },
+      issuance.assetEntropy,
+    );
+
+    // add the issuance to the input.
+    this.__CACHE.__TX.ins[inputIndex].issuance = issuance;
+
+    const kOne = Buffer.from('01', 'hex');
+    const asset = Buffer.concat([kOne, calculateAsset(entropy)]);
+    const assetScript = toOutputScript(args.assetAddress, args.net);
+
+    // send the asset amount to the asset address.
+    this.addOutput({
+      value: issuance.assetAmount,
+      script: assetScript,
+      asset,
+      nonce: Buffer.from('00', 'hex'),
+    });
+
+    // check if the token amount is not 0
+    if (args.tokenAmount !== 0) {
+      if (!args.tokenAddress)
```

Comment:
```suggestion
    if (args.tokenAmount !== 0 && !args.tokenAddress) {
```

This check must be done at the very beginning of the function because here we have already partially updated the Psbt instance and throwing an error here would cause it to be corrupted.